src/machine.rs

```diff
@@ -1093,26 +1096,84 @@ impl OlmMachine {
             stream.for_each(move |item| send_user_identities_to_callback(callback_ref, item)).await;
         });
     }
-    /// Register a callback which will be called whenever a secret is received.
+
+    /// Register a callback which will be called whenever a secret
+    /// (`m.secret.send`) is received.
+    ///
+    /// To request a secret from other devices, a client sends an
+    /// `m.secret.request` device event with action set to request and name set
+    /// to the identifier of the secret. A device that wishes to share the
+    /// secret will reply with an m.secret.send event, encrypted using olm.
+    ///
+    /// The secrets are guaranteed to have been received over a 1-to-1 encrypted
+    /// to_device message from a verified own device.
     ///
-    /// `callback` should be a function that takes 2 arguments the secret name
-    /// and value (base64) and returns a Promise.
+    /// See https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk_crypto/store/struct.Store.html#method.secrets_stream for more information.
+    ///
+    /// `callback` should be a function that takes 2 arguments the secret name.
+    /// If the secret is valid and handled on the javascript side, the secret
+    /// inbox should be cleared by calling `delete_secrets_from_inbox`.
```

Comment:
Suggest emphasising this a bit more:

```suggestion
    ///
    /// **Note**: if the secret is valid and handled on the javascript side, the secret
    /// inbox should be cleared by calling `delete_secrets_from_inbox`.
```